src/components/StoriesGroup/StoriesGroup.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import {Modal, ModalCloseReason} from '../Modal';
+import {IndexType, StoriesLayout} from '../Stories/components/StoriesLayout/StoriesLayout';
+import {StoriesPreview} from './components';
+import {StoriesGroupItem} from './types';
+
+import {block} from '../utils/cn';
+import './StoriesGroup.scss';
+
+const b = block('stories-group');
+
+export interface StoriesGroupProps {
+    open: boolean;
+    groups: StoriesGroupItem[];
+    initialStoryIndex?: [groupIndex: number, itemIndex: number];
+    disableOutsideClick?: boolean;
+    maxSliderItemsCount?: number;
+    onClose?: (
+        event: MouseEvent | KeyboardEvent | React.MouseEvent<HTMLElement, MouseEvent>,
+        reason: ModalCloseReason | 'closeButtonClick',
+    ) => void;
+    onItemSelect?: (
+        itemIndexes: [groupIndex: number, itemIndex: number],
+        selectedFromThumbnail: boolean,
+    ) => void;
+}
+
+export const StoriesGroup = ({
+    open,
+    groups,
+    onItemSelect,
+    disableOutsideClick,
+    initialStoryIndex = [0, 0],
+    maxSliderItemsCount = 12,
```

Comment:
Let's make it constant `DEFAULT_MAX_SLIDER_ITEMS_COUNT`

Reply:
I wouldn't do this because it seems that the memory area for this number will never be released on the client side.

Reply:
We don't skimp on constants, readability and clarity are a priority.

Reply:
[89ef897](https://github.com/gravity-ui/uikit/pull/387/commits/89ef897cae4b92bca252fd2c0c29a14c304cb450)